app/src/backend/UnitConversionService.ts

```diff
@@ -0,0 +1,73 @@
+// Purpose: Service to convert units between different systems.
+
+import createHttpError from "http-errors";
+
+export default class UnitConversionService {
+  private static conversionTable: { [key: string]: { [key: string]: number } } =
+    {
+      "units-cubic-meters": {
+        "units-gallons": 0.00378541, // 1 gallon = 0.00378541 cubic meters
```

Comment:
Maybe we should use decimal.js for conversions here at some point, so we don't run into floating point inaccuracy issues... This many decimals are probably handle-able by JS Numbers, but it would be nice to check if the resulting precision is sufficient in a later QA step.

Reply:
Noted